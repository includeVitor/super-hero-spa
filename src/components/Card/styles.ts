import styled, { css } from 'styled-components'

type ArticleProps = {
  $selected?: boolean
}

const Article = styled.article<ArticleProps>`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding: 1rem;
  max-width: 15rem;
  max-height: 25rem;
  overflow-y: auto;
  overflow-x: hidden;
  border-radius: 1rem;
  border: solid 0.125rem grey;
  ${props =>
    props.$selected &&
    css`
      opacity: 0.7;
    `}}
`

const Title = styled.span`
  text-align: center;
  font-weight: bold;
`

const Description = styled.span`
  text-align: center;
`

const Image = styled.img`
  width: 15rem;
  height: 20rem;
  border-radius: 1rem;
`

export { Article, Title, Description, Image }
