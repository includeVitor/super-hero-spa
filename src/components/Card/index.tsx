import { memo, useCallback, useState } from 'react'
import EmptyImage from '../EmptyImage'
import { Article, ButtonAlign, Description, Image, Title } from './styles'

type CardProps = {
  id: string
  title: string
  description?: string
  url: string
  $selected?: boolean
  handleView?: any
  handleCombat?: any
}

const Card = ({
  id,
  title,
  description,
  url,
  $selected,
  handleView,
  handleCombat
}: CardProps) => {
  const [imgLoaded, setImgLoaded] = useState(true)

  const handleImgLoadError = useCallback(() => {
    setImgLoaded(false)
  }, [])

  return (
    <Article $selected={$selected}>
      <Title>{title}</Title>
      <Description>{description}</Description>
      {imgLoaded ? (
        <Image src={url} onError={handleImgLoadError} />
      ) : (
        <EmptyImage />
      )}

      <ButtonAlign>
        {handleView && <button onClick={() => handleView(id)}>View</button>}
        {handleCombat && (
          <button onClick={() => handleCombat(id)}>Combat</button>
        )}
      </ButtonAlign>
    </Article>
  )
}

export default memo(Card)
