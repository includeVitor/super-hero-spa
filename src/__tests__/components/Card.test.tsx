import { render } from '@testing-library/react'
import Card from '../../components/Card'

describe('Card', () => {
  it('should match the snapshot card selected', () => {
    const { container } = render(
      <Card
        id="20"
        title="snapshot"
        url=""
        description="snapshot"
        handleCombat={jest.fn()}
        handleView={jest.fn()}
        $selected
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should match the snapshot card not selected', () => {
    const { container } = render(
      <Card
        id="20"
        title="snapshot"
        url=""
        description="snapshot"
        handleCombat={jest.fn()}
        handleView={jest.fn()}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
