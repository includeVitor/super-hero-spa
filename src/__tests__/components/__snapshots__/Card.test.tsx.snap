// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot card not selected 1`] = `
<article
  class="sc-gsDKAQ hUfPUh"
>
  <span
    class="sc-dkPtRN dUKHXj"
  >
    snapshot
  </span>
  <span
    class="sc-hKwDye iJFtxk"
  >
    snapshot
  </span>
  <img
    class="sc-eCImPb iRoiIz"
    src=""
  />
  <div
    class="sc-jRQBWg iLrlcE"
  >
    <button>
      View
    </button>
    <button>
      Combat
    </button>
  </div>
</article>
`;

exports[`Card should match the snapshot card selected 1`] = `
<article
  class="sc-gsDKAQ kbitaU"
>
  <span
    class="sc-dkPtRN dUKHXj"
  >
    snapshot
  </span>
  <span
    class="sc-hKwDye iJFtxk"
  >
    snapshot
  </span>
  <img
    class="sc-eCImPb iRoiIz"
    src=""
  />
  <div
    class="sc-jRQBWg iLrlcE"
  >
    <button>
      View
    </button>
    <button>
      Combat
    </button>
  </div>
</article>
`;
